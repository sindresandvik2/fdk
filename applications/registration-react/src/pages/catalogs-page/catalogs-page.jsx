import React from 'react';
import PropTypes from 'prop-types';
import { CardGroup } from 'reactstrap';
import _ from 'lodash';

import localization from '../../utils/localization';
import { Catalog } from './catalogs/catalogs.component';
import getTranslateText from '../../utils/translateText';
import './catalogs-page.scss';

const renderCatalogs = props => {
  const {
    catalogItems,
    datasets,
    apis,
    fetchDatasetsIfNeeded,
    fetchApisIfNeeded
  } = props;

  if (!catalogItems) {
    return null;
  }

  return catalogItems.map(item => (
    <div key={_.get(item, 'id')} className="row mb-2 mb-md-5">
      <div className="col-12">
        <div>
          <h2 className="fdk-text-strong mb-4">
            {getTranslateText(_.get(item, ['publisher', 'prefLabel'])) ||
              _.get(item, ['publisher', 'name'], '')}
          </h2>
        </div>
        <CardGroup>
          {datasets && (
            <Catalog
              key={`datasets-${item.id}`}
              catalogId={item.id}
              type="datasets"
              fetchItems={fetchDatasetsIfNeeded}
              items={datasets}
            />
          )}
          {apis && (
            <Catalog
              key={`apis-${item.id}`}
              catalogId={item.id}
              fetchItems={fetchApisIfNeeded}
              type="apis"
              items={apis}
            />
          )}
        </CardGroup>
      </div>
    </div>
  ));
};

export const RegCatalogs = props => {
  const { isFetching, catalogItems, fetchCatalogsIfNeeded } = props;
  const catalogsURL = '/catalogs';
  fetchCatalogsIfNeeded(catalogsURL);
  return (
    <div className="container">
      {catalogItems && renderCatalogs(props)}
      {!isFetching &&
        !catalogItems && (
          <div className="row mb-2 mb-md-5">
            <div id="no-catalogs">
              <h1 className="fdk-text-strong">
                {localization.catalogs.missingCatalogs.title}
              </h1>
              <div className="mt-2 mb-2">
                {localization.catalogs.missingCatalogs.ingress}
              </div>
              <div className="fdk-text-size-small">
                <strong>
                  {localization.catalogs.missingCatalogs.accessTitle}
                </strong>
                <p>{localization.catalogs.missingCatalogs.accessText}</p>
                <strong>
                  {localization.catalogs.missingCatalogs.assignAccessTitle}
                </strong>
                <p>{localization.catalogs.missingCatalogs.assignAccessText}</p>
              </div>
            </div>
          </div>
        )}
    </div>
  );
};

RegCatalogs.defaultProps = {
  catalogItems: null,
  isFetching: false,
  fetchCatalogsIfNeeded: _.noop
};

RegCatalogs.propTypes = {
  catalogItems: PropTypes.array,
  isFetching: PropTypes.bool,
  fetchCatalogsIfNeeded: PropTypes.func
};
