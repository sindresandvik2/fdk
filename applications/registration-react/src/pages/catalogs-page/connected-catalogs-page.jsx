import { connect } from 'react-redux';
import { fetchCatalogsIfNeeded } from '../../redux/modules/catalogs';
import { fetchDatasetsIfNeeded } from '../../redux/modules/datasets';
import { fetchApisIfNeededAction } from '../../redux/modules/apis';
import { RegCatalogs } from './catalogs-page';

function mapStateToProps({ catalogs, datasets, apis }) {
  const { catalogItems, isFetching } = catalogs || {
    catalogItems: null
  };

  return {
    catalogItems,
    datasets,
    apis,
    isFetching
  };
}

const mapDispatchToProps = dispatch => ({
  fetchCatalogsIfNeeded: catalogsURL =>
    dispatch(fetchCatalogsIfNeeded(catalogsURL)),
  fetchDatasetsIfNeeded: catalogId =>
    dispatch(fetchDatasetsIfNeeded(catalogId)),
  fetchApisIfNeeded: catalogId => dispatch(fetchApisIfNeededAction(catalogId))
});

export const ConnectedCatalogsPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(RegCatalogs);

/*
const mapStateToProps = ({ helptexts, catalog, datasets }) => {
  const { helptextItems } = helptexts || {};
  return {
    helptextItems,
    catalog,
    datasets
  };
};

const mapDispatchToProps = dispatch => ({
  fetchHelptextsIfNeeded: () => dispatch(fetchHelptextsIfNeeded()),
  fetchCatalogIfNeeded: catalogId => dispatch(fetchCatalogIfNeeded(catalogId)),
  fetchDatasetsIfNeeded: catalogId => dispatch(fetchDatasetsIfNeeded(catalogId))
});

export const ConnectedDatasetsListPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(DatasetsListPage);
  */
