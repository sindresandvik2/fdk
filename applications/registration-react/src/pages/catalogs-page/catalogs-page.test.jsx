import React from 'react';
import { shallow } from 'enzyme';
import { RegCatalogs } from './catalogs-page';
import catalogs from '../../../test/fixtures/catalogs';

let defaultProps;
let wrapper;
let dispatch;

beforeEach(() => {
  dispatch = jest.fn();
  defaultProps = {
    dispatch,
    catalogItems: catalogs.catalogItems,
    isFetchingCatalogs: false
  };
  wrapper = shallow(<RegCatalogs {...defaultProps} />);
});

test('should render RegCatalogs correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
