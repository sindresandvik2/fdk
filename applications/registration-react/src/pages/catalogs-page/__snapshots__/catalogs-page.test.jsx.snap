// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RegCatalogs correctly 1`] = `
<div
  className="container"
>
  <div
    className="row mb-2 mb-md-5"
    key="910244132"
  >
    <div
      className="col-12"
    >
      <div>
        <h2
          className="fdk-text-strong mb-4"
        >
          RAMSUND OG ROGNAN REVISJON
        </h2>
      </div>
      <CardGroup
        tag="div"
      />
    </div>
  </div>
</div>
`;
